service_discovery/samples/simple-discovery/consumer/src/main.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+//! A simple consumer for a sample of Chariott Service Discovery.
+//!
+//! This consumer "discovers" the hello world service through Chariott, and then
+//! directly calls the SayHello method on it, using a known interface. This returns
+//! a message containing "Hello, " followed by the string provided in the request.
+
+// Tells cargo to warn if a doc comment is missing and should be provided.
+#![warn(missing_docs)]
+
+use proto_servicediscovery::chariott_registry::v1::registry_client::RegistryClient;
+use proto_servicediscovery::chariott_registry::v1::DiscoverServiceRequest;
+use proto_servicediscovery::hello_world::v1::hello_world_client::HelloWorldClient;
+use proto_servicediscovery::hello_world::v1::HelloRequest;
+use tonic::Request;
+use tracing::info;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::EnvFilter;
+
+/// URL for the chariott service registry
+const CHARIOTT_SERVICE_REGISTRY_URL: &str = "http://0.0.0.0:50000";
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Set up tracing
+    let collector = tracing_subscriber::fmt()
+        .with_env_filter(
+            EnvFilter::builder()
+                .with_default_directive(tracing::Level::INFO.into())
+                .from_env_lossy(),
+        )
+        .finish();
+
+    collector.init();
+
+    // Create a registry client
+    let mut registry_client = RegistryClient::connect(CHARIOTT_SERVICE_REGISTRY_URL).await?;
+    let discover_request = Request::new(DiscoverServiceRequest {
+        namespace: String::from("sdv.samples"),
+        name: String::from("hello-world"),
+        version: String::from("1.0.0.0"),
+    });
+
+    // Discover the simple provider service
+    let service_option =
+        registry_client.discover_service(discover_request).await?.into_inner().service;
+    match service_option {
+        Some(service) => {
+            info!("Discovered service {:?}", service);
+            if service.communication_kind != *"grpc+proto"
```

Comment:
Please consider making communication_kind an enum, and then you can use these 2 crates: `strum` (https://docs.rs/strum/latest/strum/)  and `strum_macros`.

Specifically, you may want to use the EnumString attribute from the strum_macros crate (https://docs.rs/strum/latest/strum/derive.EnumString.html), which allows you to match a String to an enum variant